worker/instancemutater/worker_test.go

```diff
@@ -189,72 +185,92 @@ var _ = gc.Suite(&workerEnvironSuite{})
 // below to compose a test of the whole instance mutator scenario, from start
 // to finish for an EnvironWorker.
 func (s *workerEnvironSuite) TestFullWorkflow(c *gc.C) {
-	defer s.setup(c).Finish()
+	defer s.setup(c, 1).Finish()
 
 	w := s.workerForScenario(c,
 		s.ignoreLogging(c),
-		s.notifyMachines([][]string{
-			{"0"},
-		}, s.noopDone),
-		s.expectFacadeMachineTag,
-		s.notifyMachineAppLXDProfile(1, s.closeDone),
-		s.expectMachineTag,
-		s.expectMachineCharmProfilingInfo(3),
-		s.expectLXDProfileNames,
-		s.expectSetCharmProfiles,
+		s.notifyMachines([][]string{{"0"}}),
+		s.expectFacadeMachineTag(0),
+		s.notifyMachineAppLXDProfile(0, 1),
+		s.expectMachineTag(0),
```

Comment:
Looks like `expectFacadeMachineTag` already satisfies this.

Reply:
removed